src/CSharpLanguageServer/Server.fs

```diff
@@ -852,6 +856,36 @@ type CSharpLspServer(lspClient: CSharpLspClient, options: Options) =
 
         async { return metadataMaybe |> success }
 
+    override __.TextDocumentFormatting(format: Types.DocumentFormattingParams) : AsyncLspResult<Types.TextEdit [] option> =
+        async {
+            let maybeDocument = getDocumentForUri state format.TextDocument.Uri
+            let! formattingChanges =
+                match maybeDocument with
+                | Some doc -> handleTextDocumentFormatAsync doc
+                | None -> async { return Array.empty<Types.TextEdit> }
+            return formattingChanges |> Some |> success
+        }
+
+    override __.TextDocumentRangeFormatting(format: DocumentRangeFormattingParams) : AsyncLspResult<TextEdit[] option> =
+        async {
+             let maybeDocument = getDocumentForUri state format.TextDocument.Uri
+             let! formattingChanges =
+                 match maybeDocument with
+                 | Some doc -> handleTextDocumentRangeFormatAsync doc format.Range
+                 | None -> async { return Array.empty<Types.TextEdit> }
+             return formattingChanges |> Some |> success
+        }
+
+    override __.TextDocumentOnTypeFormatting(format: DocumentOnTypeFormattingParams) : AsyncLspResult<TextEdit [] option> =
+        async {
+            let maybeDocument = getDocumentForUri state format.TextDocument.Uri
+            let! formattingChanges =
+                match maybeDocument with
+                | Some doc -> handleTextOnTypeFormatAsync doc format.Ch format.Position
+                | None -> async { return Array.empty<Types.TextEdit> }
```

Comment:
I think you can just do
```
| None -> async { return [] }
```

or maybe there is an even better way (but I am newbie in F#).. maybe there is some helper to do this
```
| None -> [] |> Async.Result
```

Reply:
same thing on other places -- no need to to `Array.empty<>` -- F# uses type inference more than C# and you can just type `[]` in those places

Reply:
So thats what I thought too but the compiler throws `error FS0001: This expression was expected to have type    'TextEdit[]'    but here has type    ''a list` with the 
```
| None -> async { return [] }
```
syntax